app/scripts/lib/util.ts

```diff
@@ -235,10 +235,43 @@ export function previousValueComparator<A>(
 }
 
 export function addUrlProtocolPrefix(urlString: string) {
-  if (!urlString.match(/(^http:\/\/)|(^https:\/\/)/u)) {
-    return `https://${urlString}`;
+  let trimmed = urlString.trim();
+
+  if (trimmed.length && !urlLib.parse(trimmed).protocol) {
+    trimmed = `https://${trimmed}`;
   }
-  return urlString;
+
+  if (getValidUrl(trimmed) !== null) {
```

Comment:
It looks like we need to add additional tests for edge cases like this 

```
..
.asdf.
.asdf
adsf.
```
  If we prepend the http:// before getValidUrl these cases would be valid.

Reply:
As discussed with @HowardBraham, we can handle these edge cases in another pr.